vm/src/vm/instruction/execution.rs

```
@@ -16,53 +16,73 @@ impl Instruction {
     ) -> Result<Log, ExecutionError> {
         println!("registers: {:?}", &registers);
         println!("Executing instruction at 0x{:08x}: {:?}", *pc, self);
-        let (new_pc, updated_register, new_register_value) =
-            self.execute(*pc, registers, memory)?;
-        *pc = new_pc;
-        if updated_register != 0 {
-            registers.0[updated_register as usize] = new_register_value;
-        }
-        Ok(Log {
-            instruction: self,
-            updated_register_value: new_register_value,
-            updated_pc: *pc,
-        })
+        let log = self.execute(*pc, registers, memory)?;
+        // Cleanup zero register in case it was written to
+        registers.0[0] = 0;
```

Shouldn't we return an error here? It would be a bug

---

Added TODO comment!